compiler/src/Language/Bond/Codegen/Java/Class_java.hs

```diff
@@ -10,25 +10,216 @@ module Language.Bond.Codegen.Java.Class_java
+        javaStructFieldDescriptorInitTypeExpr BT_UInt64 = [lt|com.microsoft.bond.BondTypes.UINT64|]
+        javaStructFieldDescriptorInitTypeExpr BT_Float = [lt|com.microsoft.bond.BondTypes.FLOAT|]
+        javaStructFieldDescriptorInitTypeExpr BT_Double = [lt|com.microsoft.bond.BondTypes.DOUBLE|]
+        javaStructFieldDescriptorInitTypeExpr BT_Bool = [lt|com.microsoft.bond.BondTypes.BOOL|]
+        javaStructFieldDescriptorInitTypeExpr BT_String = [lt|com.microsoft.bond.BondTypes.STRING|]
+        javaStructFieldDescriptorInitTypeExpr BT_WString = [lt|com.microsoft.bond.BondTypes.WSTRING|]
+        javaStructFieldDescriptorInitTypeExpr BT_Blob = [lt|com.microsoft.bond.BondTypes.BLOB|]
+        javaStructFieldDescriptorInitTypeExpr (BT_Bonded t) = [lt|bondedOf(#{javaStructFieldDescriptorInitTypeExpr t})|]
+        javaStructFieldDescriptorInitTypeExpr (BT_Nullable t) = [lt|nullableOf(#{javaStructFieldDescriptorInitTypeExpr t})|]
+        javaStructFieldDescriptorInitTypeExpr (BT_Vector t) = [lt|vectorOf(#{javaStructFieldDescriptorInitTypeExpr t})|]
+        javaStructFieldDescriptorInitTypeExpr (BT_List t) = [lt|listOf(#{javaStructFieldDescriptorInitTypeExpr t})|]
+        javaStructFieldDescriptorInitTypeExpr (BT_Set t) = [lt|setOf(#{javaStructFieldDescriptorInitTypeExpr t})|]
+        javaStructFieldDescriptorInitTypeExpr (BT_Map k v) = [lt|mapOf(#{javaStructFieldDescriptorInitTypeExpr k}, #{javaStructFieldDescriptorInitTypeExpr v})|]
+        javaStructFieldDescriptorInitTypeExpr (BT_TypeParam param) = [lt|#{paramName param}|]
+        javaStructFieldDescriptorInitTypeExpr (BT_UserDefined e@Enum {..} _) = [lt|#{qualifiedDeclaredTypeName java e}.BOND_TYPE|]
+        javaStructFieldDescriptorInitTypeExpr (BT_UserDefined s@Struct {..} params) = [lt|#{structFieldDescriptorInitStructExpr java (qualifiedDeclaredTypeName java s) params}|]
+        javaStructFieldDescriptorInitTypeExpr (BT_UserDefined s@Forward {..} params) = [lt|#{structFieldDescriptorInitStructExpr java (qualifiedDeclaredTypeName java s) params}|]
+        javaStructFieldDescriptorInitTypeExpr t = error $ "invalid declaration type for javaFieldBondTypeExpression: " ++ (show t)
+
+
+-- returns an type descriptor expression for a given struct base
+structBaseDescriptorInitStructExpr :: MappingContext -> (Maybe Type) -> Text
+structBaseDescriptorInitStructExpr _ Nothing = [lt|null|]
+structBaseDescriptorInitStructExpr java (Just t) = structFieldDescriptorInitTypeExpr java t
+
+
+-- returns an expression for the default value of a field that has an explicit default, used in intiialization of struct field descriptors
+fieldDefaultValueInitParamExpr :: MappingContext -> Type -> Maybe Default -> Text
+fieldDefaultValueInitParamExpr java type_ default_ = javaFieldDefaultValueInitParamExpr type_ default_
+    where
+        javaFieldDefaultValueInitParamExpr _ (Just (DefaultBool val)) = if val
+            then [lt|true|]
```

Comment:
We shouldn't rely on Haskell's string constants to generate Java. Further, in Haskell they are capitalized and in Java are lowercase.

---
In reply to: [125134970](https://github.com/Microsoft/bond/pull/517#discussion_r125134970) [](ancestors = 125134970)